### ISSUE TITLE
Wrong type to $in expr

### ISSUE THREAD

**Issue by yuval9313** [Automattic/mongoose#11949]
### Prerequisites

- [X] I have written a descriptive issue title

### Mongoose version

6.4.0

### Node.js version

16.14.2

### MongoDB version

cosmosDB 4 - not really relevant

### Operating system

Windows

### Operating system version (i.e. 20.04, 11.3, 10)

WSL 2 - Ubuntu 20.04

### Issue

I have received this issue 

`Type 'string[]' is not assignable to type '[Expression, ArrayExpression<any>]'. Target requires 2 element(s) but source may have fewer.`

for this snippet:
```js
{
    $match: {
        'data.sourceCurrency': { $in: ['USD', 'GBP', 'EUR'] },
    },
}
```

I'm not sure what more data to provide, but unless I've mistaken it looks like the $in an array is an array of 2 items which `[Expression, ArrayExpression<any>]` instead of `Expression[] | ArrayExpression<any>[]`

P.S: `ArrayExpression`  covers `T[]` which includes `string[]` which is my type

**Comment by Uzlopak** [comment]
This is some inconsistency between the docs of mongo and real mongodb syntax. I will provide a fix tomorrow. I am currently not having access to my Laptop.

**Comment by heilmela** [comment]
im facing similiar problems with some aggregate pipelines
```  
const pipeline: PipelineStage[] = [
      {
        $unwind: {
          path: '$sectors',
          preserveNullAndEmptyArrays: true,
        },
      },
      {
        $match: {
          organization,
          approved: { $exists: true, $ne: null },
        },
      }
  ]
 ```
 
Where i get `Type 'null' is not assignable to type 'Expression | [Expression, Expression]'`. Removing the `$ne` leaves me with `Type '{ $exists: true; }' is not assignable to type 'Expression'.`

**Comment by sm-hejazi** [comment]
+1

**Comment by ezrabowman** [comment]
Seeing this also with:
```
UserModel.aggregate([{ $match: { name: { "$exists": false } } }])

Type '{ $exists: false; }' is not assignable to type 'Expression'.
```

This is new in 6.4.0 - works fine on 6.3.9

**Comment by vkarpov15** [comment]
This should be fixed by #11969
